appinventor/appengine/src/com/google/appinventor/client/actions/SendToGalleryAction.java

```diff
@@ -0,0 +1,60 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2009-2011 Google, All Rights reserved
+// Copyright 2011-2023 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.client.actions;
+
+import com.google.appinventor.client.ErrorReporter;
+import com.google.appinventor.client.Ode;
+import com.google.appinventor.client.OdeAsyncCallback;
+import com.google.appinventor.client.boxes.ProjectListBox;
+import com.google.appinventor.client.explorer.project.Project;
+import com.google.appinventor.shared.rpc.RpcResult;
+import com.google.gwt.user.client.Command;
+import com.google.gwt.user.client.Window;
+
+import java.util.List;
+
+import static com.google.appinventor.client.Ode.MESSAGES;
+
+// Send to the New Gallery
+public class SendToGalleryAction implements Command {
+  private static volatile boolean lockPublishButton = false; // To prevent double clicking
+
+  @Override
+  public void execute() {
+    if (Ode.getInstance().getCurrentView() == Ode.PROJECTS) {
```

Comment:
Note that this logic is duplicated from editor/youngandroid/actions/SendToGalleryAction. We may just want to make this one subclass that one so we don't duplicate the gallery logic (in case it has to change in the future.